index.js

```diff
@@ -0,0 +1,180 @@
+     * @param  {String}   config.container  Container for the build to run in
+     * @param  {String}   config.scmUrl     Scm URL to use in the build
+     * @param  {Function} callback          Callback with null if successful otherwise error
+     */
+    _start(config, callback) {
+        const fakeConfigPath = path.resolve(__dirname, './config/test-job.xml');
+        const xml = fs.readFileSync(fakeConfigPath, 'utf-8');
+
+        async.waterfall([
+            this.getCrumb.bind(this),
+            this.initJenkinsClient.bind(this),
+            (jenkinsClient, cb) => {
+                const create = jenkinsClient.job.create.bind(jenkinsClient.job);
+
+                return create(config.buildId, xml, (err) => {
+                    cb(err, jenkinsClient);
+                });
+            },
+            (jenkinsClient, cb) => {
+                const build = jenkinsClient.job.build.bind(jenkinsClient.job);
+
+                return build(config.buildId, cb);
+            }
+        ], (error) => {
+            if (error) return callback(new Error(error.message));
+
+            return callback(null);
+        });
+    }
+
+    /**
+     * Stop the build
+     * @method createJob
+     * @param  {Object}   config            A configuration object
+     * @param  {String}   config.buildId    ID for the build and also name of the job in jenkins
+     * @param  {Function} callback          Callback with null if successful otherwise error
+     */
+    _stop(config, callback) {
+        async.waterfall([
+            this.getCrumb.bind(this),
+            this.initJenkinsClient.bind(this),
+            (jenkinsClient, cb) => {
+                const get = jenkinsClient.job.get.bind(jenkinsClient.job);
+
+                return get(config.buildId, (err, data) => {
+                    try {
+                        cb(err, jenkinsClient, data.lastBuild.number);
+                    } catch (e) {
+                        cb(e);
+                    }
+                });
+            },
+            (jenkinsClient, buildNumber, cb) => {
+                const stop = jenkinsClient.build.stop.bind(jenkinsClient.build);
+
+                return stop(config.buildId, buildNumber, cb);
+            }
+        ], (error) => {
+            if (error) return callback(new Error(error.message));
+
+            return callback(null);
+        });
+    }
+
+    /**
+    * Streams logs
+    * @method stream
+    * @param  {Object}   config            A configuration object
+    * @param  {String}   config.buildId    ID for the build
+    * @param  {Response} callback          Callback with error and data
+    */
+    _stream(config, callback) {
+        async.waterfall([
+            this.getCrumb.bind(this),
+            this.initJenkinsClient.bind(this),
+            (jenkinsClient, cb) => {
+                const get = jenkinsClient.job.get.bind(jenkinsClient.job);
+
+                return get(config.buildId, (err, data) => {
```

Comment:
This block can be refactored into a separate function that since it's being used above as well